Merge release 1.35.0 to develop

cameronvoell opened the pull request:
Merging the 1.35.0 release from main to develop.

Checks before merging this PR:

- [x] Updated the submodule hash for gutenberg after corresponding gutenberg PR is merged: https://github.com/WordPress/gutenberg/pull/24773


PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/docs/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

cameronvoell committed:
Release script: Update gb mobile version to 1.34.1

cameronvoell committed:
Release script: Update bundle for: 1.34.1

cameronvoell committed:
Update release notes and gutenberg ref

cameronvoell committed:
Update gutenberg ref

cameronvoell committed:
Update bundle

cameronvoell committed:
Merge pull request #2556 from wordpress-mobile/release/1.34.1

Hotfix Release 1.34.1

cameronvoell committed:
Release script: Update gb mobile version to 1.35.0

cameronvoell committed:
Release script: Update bundle for: 1.35.0

cameronvoell committed:
Updated gutenberg ref

cameronvoell committed:
Updated bundle

cameronvoell committed:
Merge branch 'main' into release/1.35.0 and update
bundle one more time

cameronvoell committed:
updated release notes

cameronvoell committed:
Merge pull request #2557 from wordpress-mobile/release/1.35.0

Release 1.35.0

cameronvoell committed:
Update gutenberg ref

ceyhun reviewed (approved):
`Check Correctness` CI job is failing but seems like it's not related to this.

cameronvoell committed:
Merge branch 'develop' into merge_release_1.35_to_develop

cameronvoell committed:
Updated gutenberg ref

cameronvoell committed:
Merge branch 'develop' into merge_release_1.35_to_develop

cameronvoell commented:
Looking for one more review on this PR from an Android perspective. Perhaps @mchowning  can take a look. 

Everything in this PR looks standard except I noticed in the PR that when updating `bundle/` dir during the release process I deleted `hermes.aar` files in the gb-mobile bundle directory unintentionally ([commit for reference](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2568/commits/a1d2aee9afb0f24bd3572786aed60f047c0e2322) - I'm thinking I deleted bundle dir and ran `npm run bundle` to resolve conflicts). I think why this went unnoticed is because those files have not been used since Monorepo. See commits below for reference ([rnmobile/monorepo-commit-history](https://github.com/WordPress/gutenberg/tree/rnmobile/monorepo-commit-history) tag was useful for verifying this). 

Commit when hermes.aar files were added to be utilized by the RN bridge: https://github.com/wordpress-mobile/gutenberg-mobile/commit/9a044fe717e6410a743fbb1f74f0ecc9f09e25f0#diff-fbf0c7c1b326fd22137e34cdded191f1

Commit when hermes.aar files were added in the react-native-bridge package on gutenberg: https://github.com/WordPress/gutenberg/commit/37c76c46a2c365353fec466ec855584f0c7a4180#diff-f3eaa5453261deb18f2878c75fd3a10b

I added a line to the description of this PR about the removal of the unnecessary hermes.aar files. Does this look good to you @mchowning?

mchowning commented:
> I noticed in the PR that when updating bundle/ dir during the release process I deleted hermes.aar files in the gb-mobile bundle directory unintentionally.... I think why this went unnoticed is because those files have not been used since Monorepo. 

I did some digging too and I cannot find anything that is trying to use those hermes files in the gb-mobile bundle/ directory, so I agree that removing them was actually a good thing. 😄

cameronvoell committed:
Merge branch 'develop' into merge_release_1.35_to_develop
